spec/factories/notification.rb

```diff
@@ -1,7 +1,11 @@
 FactoryBot.define do
   factory :notification do
     association :subscription
-    association :notification_type, :webhook # defaults to webhook
+
+    notification_type do
+      NotificationType.find_or_create_by(id: 'webhook', title: 'Webhook')
```

Comment:
NotificationType is a table with two entries in it. The primary keys chosen are one of: 'email' or 'webhook'. This is so far out of the standard approach to primary keys in rails that I'm having to do some special snowflake wrapper around the association as this approach isn't compatible with default factory bot rails approaches to handling relationships.